website/blog/testing.md

```diff
@@ -0,0 +1,604 @@
+example, we can write a fake implementation of the product catalog service that
+uses a fixed in-memory set of products:
+
+```go
+type Product = productcatalogservice.Product
+
+type fakeCatalog struct {
+    products []Product
+}
+
+func (f *fakeCatalog) ListProducts(context.Context) ([]Product, error) {
+    return f.products, nil
+}
+
+func (f *fakeCatalog) GetProduct(ctx context.Context, id string) (Product, error) {
+    for _, p := range f.products {
+        if p.ID == id {
+            return p, nil
+        }
+    }
+    return Product{}, productcatalogservice.NotFoundError{}
+}
+
+func (f *fakeCatalog) SearchProducts(context.Context, string) ([]Product, error) {
+	panic("unimplemented")
+}
+```
+
+Then, we can update our integration test to use a `fakeCatalog` instead of the
+real product catalog service component.
+
+```go
+func TestPurchaseWithFakeCatalog(t *testing.T) {
+    catalog := &fakeCatalog{
+        products: []Product{
+            {
+                ID:          "0",
+                Name:        "Noogler Hat",
+                Description: "A colorful hat given to new Googlers.",
+                PriceUSD:    money.T{CurrencyCode: "USD", Units: 42},
+            },
+        },
+    }
+    fake := weavertest.Fake[productcatalogservice.ProductCatalogService](catalog)
+    runner := weavertest.Local
+    runner.Fakes = append(runner.Fakes, fake)
+    runner.Test(t, func(
+        t *testing.T,
+        catalog productcatalogservice.ProductCatalogService,
+        cart cartservice.CartService,
+        checkout checkoutservice.CheckoutService,
+    ) {
+        // Testing code goes here.
+        ...
+    })
+}
+```
+
+## 3. Randomized Testing
+
+When you write a unit test or integration test, you write some code and check
+that it executes in the way you expect. For example, to test a `Reversed([]byte)
+[]byte` function, you might check that
+
+- `Reversed([]byte{})` is `[]byte{}`,
+- `Reversed([]byte{1})` is `[]byte{1}`,
+- `Reversed([]byte{1, 2})` is `[]byte{2, 1}`, and so on.
+
+Unit and integration tests are thus limited to only test the behaviors and
+corner cases that you can think of and have the patience to write down.
+
+With randomized **property-based tests**, on the other hand, you specify a
+property of your code that should always hold and then test the property on
+millions of randomly generated examples. For example, we can test that the
+`Reversed` function is [involutive][]&mdash;i.e. that `Reversed(Reversed(x)) ==
+x`&mdash;by writing a [fuzz test][fuzzing] that tests the property on randomly
+generated byte slices:
+
+```go
+func FuzzReversedIsInvolutive(f *testing.F) {
+    f.Fuzz(func(t *testing.T, want []byte) {
+        got := Reversed(Reversed(want))
+        if !slices.Equal(want, got) {
+            t.Fatalf("got %v, want %v", got, want)
+        }
+    })
+}
+```
+
+Service Weaver allows you to apply randomized property-based testing to entire
```

Comment:
Can we mention that the existing fuzz testing mechanisms are limited though and Service Weaver allows you test more things?

Reply:
I reworded this bit to make it clearer that Service Weaver is implementing a type of randomized testing more sophisticated that regular fuzz testing.